classes/local.php

```diff
@@ -279,26 +279,27 @@ public static function file_is_public(string $contenthash): bool {
 
         $minduplicates = get_config('filter_imageopt', 'minduplicates');
         $publicfilescache = \cache::make('filter_imageopt', 'public_files');
-        $key = 'public_files_' . $minduplicates;
+        $key = 'publicfiles' . '_' . $minduplicates;
 
         if (empty($minduplicates)) {
             return false;
         }
 
-        if (!$publicfilescache->has($key)) {
-            $publicfiles = $DB->get_records_sql(
-                "SELECT contenthash, COUNT(contenthash)
-                   FROM {files}
-                  WHERE filearea <> 'draft'
-                    AND (filearea <> 'public' OR component <> 'filter_imageopt')
-               GROUP BY contenthash
-                 HAVING COUNT(contenthash) >= :count",
-                ['count' => $minduplicates]
+        if (!$publicfiles = $publicfilescache->get($key)) {
+            $publicfilescache->set($key, []);
```

Comment:
it would be better to set $publicfiles instead of the cache here so we don't set it twice